networkx/algorithms/clique.py

```diff
@@ -540,3 +540,159 @@ def cliques_containing_node(G, nodes=None, cliques=None):
+                    raise ValueError(err.format(weight, v))
+            self.vtx_weights = {v: G.nodes[v][weight] for v in G.nodes()}
+
+    def update_incumbent_if_improved(self, C, C_weight):
+        """Update the incumbent if the vertex set C has greater weight.
+
+        C is assumed to be a clique.
+        """
+        if C_weight > self.incumbent_weight:
+            self.incumbent_vertices = C[:]
+            self.incumbent_weight = C_weight
+
+    def greedily_find_independent_set(self, P):
+        """Greedily find an independent set of vertices from a set of
+        vertices P."""
+        independent_set = []
+        P = P[:]
+        while P:
+            v = P[0]
+            independent_set.append(v)
+            P = [w for w in P if v != w and not self.G.has_edge(v, w)]
+        return independent_set
+
+    def find_branching_vertices(self, P, target):
+        """Find a set of vertices to branch on."""
+        residual_wt = {v: self.vtx_weights[v] for v in P}
+        total_wt = 0
+        P = P[:]
+        while P:
+            independent_set = self.greedily_find_independent_set(P)
+            min_wt_in_class = min(residual_wt[v] for v in independent_set)
+            total_wt += min_wt_in_class
+            if total_wt > target:
+                break
+            for v in independent_set:
+                residual_wt[v] -= min_wt_in_class
+            P = [v for v in P if residual_wt[v] != 0]
+        return P
+
+    def expand(self, C, C_weight, P):
+        """Look for the best clique that contains all the vertices in C and zero or
+        more of the vertices in P, backtracking if it can be shown that no such
+        clique has greater weight than the incumbent.
+        """
+        self.update_incumbent_if_improved(C, C_weight)
+        branching_vertices = self.find_branching_vertices(
+            P, self.incumbent_weight - C_weight)
+        while branching_vertices:
+            v = branching_vertices.pop()
+            P.remove(v)
+            new_C = C + [v]
+            new_C_weight = C_weight + self.vtx_weights[v]
+            new_P = [w for w in P if self.G.has_edge(v, w)]
+            self.expand(new_C, new_C_weight, new_P)
+
+    def find_max_weight_clique(self):
+        """Find a maximum weight clique."""
+        # Sort vertices in reverse order of degree for speed
+        vertices = sorted(self.G.nodes(), key=lambda v: self.G.degree(v),
+                          reverse=True)
+        vertices = [v for v in vertices if self.vtx_weights[v] > 0]
+        self.expand([], 0, vertices)
+
+
+@not_implemented_for('directed')
+def max_weight_clique(G, weight='weight'):
+    """Find a maximum weight clique in G.
+
+    A _clique_ in a graph is a set of vertices such that every two distinct
+    vertices are adjacent.  The _weight_ of a clique is the sum of the weights
+    of its vertices.  A _maximum weight clique_ of graph G is a clique C in G
```

Comment:
The doc isn't clear whether the weight is for edges or nodes here.

Reply:
Thanks for spotting this. Referring to edge attributes rather than node attributes was a bad mistake in my documentation, which is now fixed. (I had copied the docstring for the `weight` parameter from another NetworkX function for consistency, but had not noticed that that one used edge weights.) To algorithm is for max node weight clique, rather than the less common max edge weight clique problem.

Reply:
Probably just a nit-pick...,  but could we change "vertices" to "nodes" and vertex to node?

Reply:
Sorry - I should have done that from the start for consistency with the rest of NetworkX. I have now changed vertex/vertices to node/nodes throughout the MWC code.

Reply:
Thanks very much for your quick response.....
This looks ready!